### DISCUSSION TITLE
Implementation of MFront code for crystal plasticity in Ansys

### DISCUSSION THREAD

**Discussion by neelammishra** [thelfer/tfel#109] [Q&A]
How can I add MFront code for crystal plasticity in Ansys for microstructural analysis?

**Comment by thelfer** [comment]
Hi @neelammishra,

There are a large number of models for crystal plasticity. You may have a look at the Meric Cailletaud example for an introduction: <https://thelfer.github.io/tfel/web/MericCailletaudSingleCrystalPlasticity.html>

Concerning the `ansys` interface, it is documented here: <https://thelfer.github.io/tfel/web/ansys.html>

**Comment by jeffwitz** [comment]
I think, that the first step is to describe what kind of slip system is needed !
You can find the information on  : Definition of the crystal structure, the slip systems and the interaction matrix
that leads to the page : http://tfel.sourceforge.net/singlecrystal.html
I'm clearly not an expert, I think Thomas can provide you more information on this subject

**Reply by thelfer** [reply]
@neelammishra Have you found the information that you need ,

**Reply by neelammishra** [reply]
No. I need to define a usermat in ansys for Kocks-Mecking model. How do I do that??

**Reply by thelfer** [reply]
Well, you first need to define the Kocks-Mecking model to see how hard it would be to implement it in MFront.
